### PR TITLE
condense two for loop in port_allocator as one

### PR DESCRIPTION
**Pull request by mqliang**
This change is [Reviewable](http://reviewable.k8s.io/reviews/kubernetes/kubernetes/20321)

### CODE REVIEW

**Comment by thockin** [comment]
@jdef @karlkfi @sttts This is in mesos-related code - can you PTAL?  I'm buried :)

File: pkg/proxy/userspace/port_allocator.go [hunk 1]
```diff
@@ -118,18 +118,14 @@ func (r *rangeAllocator) nextFreePort() int {
...
-		if b := r.used.Bit(i); b == 0 {
-			r.used.SetBit(&r.used, i, 1)
-			return i + r.Base
-		}
-	}
-	for i := 0; i < j; i++ {
-		if b := r.used.Bit(i); b == 0 {
-			r.used.SetBit(&r.used, i, 1)
-			return i + r.Base
+	for i := 0; i < r.Size; i++ {
```
Comment (jdef) [inline pkg/proxy/userspace/port_allocator.go]: if we're really determined to combine the loops, then we should probably get rid of the `if` above as well:
```go
	// search sequentially
	for i := j; i < r.Size + j; i++ {
		next := i % r.Size
		if r.used.Bit(next) == 0 {
			r.used.SetBit(&r.used, next, 1)
			return next + r.Base
		}
	}
```

**Comment by jfrazelle** [comment]
this looks good to me

**Comment by thockin** [comment]
Is there a test that covers this?  It has crossed from cleanup into "I don't want to convince myself that's correct". :)

**Pull request closed without merging**